### PR TITLE
Typescript refactor, component namespace and more

### LINKED ISSUE #32: The Icon conflicts with IDE (PHPStrom/WebStorm)
**Version**
0.2.0

In PHPStrom and WebStorm your component `Icon` in conflict with my component `Icon` in my project. 
I mean, I am prompted to import your `Icon`, instead of mine.

image

This is a pretty nasty problem for me and my team. You could just rename the component, please.

### LINKED ISSUE #36: Rewrite plugin in Typescript
Rewrite the plugin using typescript + vue to make it easier to update to Vue 3 later on.

### PR DESCRIPTION
**Pull request by Maronato**
1.0 release PR

Release changes:

## ⚠️ Breaking changes
- **`hideCloseButton` removed!** Now set `closeButton: false` to hide it. (#35) (ea1a448dab458c8ca76cdf02c45ac935bee5f127)
- **Custom icon object shape changed!** Previous shape `{ tag, children, class }` conflicts with JSX, so it was changed to `{ iconTag, iconChildren, iconClass }` (#35) (ea1a448dab458c8ca76cdf02c45ac935bee5f127)
- **Renamed `examples` folder to `public`!** *Examples* did not make sense since it was the code for the demo. (#35) (eb56db27dfa33460d7310d43f4cafde41dc4485d)
- **Namespaced components with `Vt` prefix!** Some editors were picking up internal components by name. (#35) (4158311fc630045380b75d60e20c12fcc9dc5f5a) closes #32

## 🚀 New features
- **Full typescript:** Entirely rewritten in Typescript (#35) (eb56db27dfa33460d7310d43f4cafde41dc4485d), closes #36
- **Added `containerClassName` plugin option**: Now custom classes can be added to containers [docs](https://github.com/Maronato/vue-toastification/blob/58c1d3ee2d778dd12f27e74f95085a32d5913352/README.md#custom-toast-container-classes) (#35) (58c1d3ee2d778dd12f27e74f95085a32d5913352)

##  🔧 Bug fixes
- **Added Plugin options declaration to types**: `PluginOptions` was missing from the module declaration. #34 (03cd1403bb7c7f240acdca79c522edaa0cb159e7)
- **Dynamic container element updates:** `$updateDefaults({ container: yourHTMLElement } )` was not updating the container component, but now does without losing any state or affecting UI. (#35)(642834473010d95d9b50e5d99b964ad669f2b30d) Thanks to @tomeus 

## 📖 Documentation
- **Added `onClose` documentation**: `onClose` documentation was missing, even though it was implemented. [docs](https://github.com/Maronato/vue-toastification/blob/eb56db27dfa33460d7310d43f4cafde41dc4485d/README.md#toast-options-object) (#35) (eb56db27dfa33460d7310d43f4cafde41dc4485d)

##  📦 Bundle
- **Switch from webpack to rollup:** Smaller and more diverse builds with support for tree shaking. (#35) (eb56db2)

### CODE REVIEW

File: src/components/VtCloseButton.vue [hunk 1]
```diff
@@ -3,34 +3,31 @@
...
   props: PROPS.CLOSE_BUTTON,
-  data() {
-    return {
-      VT_NAMESPACE
-    };
-  },
+
   computed: {
-    buttonComponet() {
+    buttonComponet(): RenderableToastContent {
```
Comment (tomeus) [inline src/components/VtCloseButton.vue]: Typo buttonComponet => buttonComponent

File: .eslintrc.js [hunk 2]
```diff
@@ -1,45 +1,26 @@
...
-    "vue/key-spacing": "error",
-    "vue/match-component-file-name": "error",
-    "vue/object-curly-spacing": "error",
-    "vue/require-direct-export": "error",
-    "vue/script-indent": "error",
-    "vue/space-infix-ops": "error",
-    "vue/space-unary-ops": "error",
-    "vue/v-on-function-call": "error",
+    "@typescript-eslint/no-use-before-define": "off",
+    "@typescript-eslint/no-empty-function": "off",
```
[the remark this answers is not in the document]
Comment (Maronato) [reply .eslintrc.js]: It was being used, but not anymore, so you are correct in having it removed.

File: src/components/VtToastContainer.vue [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  components: { Toast, Transition },
+
+  props: Object.assign({}, PROPS.CORE_TOAST, PROPS.CONTAINER),
+
+  data() {
+    const data: {
+      count: number;
+      positions: POSITION[];
+      toasts: {
+        [toastId: number]: ToastOptions;
```
[the remark this answers is not in the document]
Comment (Maronato) [reply src/components/VtToastContainer.vue]: `[toastId: ToastID]: ToastOptions;` throws *An index signature parameter type cannot be a union type. Consider using a mapped object type instead.*, so I had to split it into `ToastID`s discrete types

File: src/components/VtCloseButton.vue (hunk 1 above)
Comment (Maronato) [reply src/components/VtCloseButton.vue]: nice catch

**Pull request merged**